From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Javier Martinez Canillas <javierm@redhat.com>
Date: Thu, 10 May 2018 10:52:11 +0200
Subject: [PATCH] Add version field to BLS generated by grub2-switch-to-blscfg

The version field is present in the BLS fragments that are shipped in the
kernel packages, so add it to the BLS generated by grub2-switch-to-blscfg
for consistency.

Signed-off-by: Javier Martinez Canillas <javierm@redhat.com>
---
 util/grub-switch-to-blscfg.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/util/grub-switch-to-blscfg.in b/util/grub-switch-to-blscfg.in
index bc28053cd30..89487ad611c 100644
--- a/util/grub-switch-to-blscfg.in
+++ b/util/grub-switch-to-blscfg.in
@@ -190,12 +190,14 @@ mkbls() {
     local datetime=$1 && shift
 
     local debugname=""
+    local debugid=""
     local flavor=""
 
     if [[ "$kernelver" == *\+* ]] ; then
         local flavor=-"${kernelver##*+}"
         if [[ "${flavor}" == "-debug" ]]; then
             local debugname=" with debugging"
+            local debugid="-debug"
         fi
     fi
     (
@@ -203,6 +205,7 @@ mkbls() {
 
         cat <<EOF
 title ${NAME} (${kernelver}) ${VERSION}${debugname}
+version ${kernelver}${debugid}
 linux /vmlinuz-${kernelver}
 initrd /initramfs-${kernelver}.img
 options \$kernelopts
